cloudferry/lib/os/actions/check_copy_engine_configuration.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2016 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+import logging
+import os
+
+from cloudferry.lib.base.action import action
+from cloudferry.lib.base import exception
+
+LOG = logging.getLogger(__name__)
+
+
+class CheckCopyEngineConfiguration(action.Action):
+    def run(self, **kwargs):
+        copy_backend = self.cfg.migrate.copy_backend
+        if copy_backend == 'bbcp':
+            self.check_bbcp()
+        elif copy_backend == 'rsync':
+            self.check_rsync()
+
+    def check_bbcp(self):
+        path = {self.cfg.bbcp.path, self.cfg.bbcp.src_path,
+                self.cfg.bbcp.dst_path}
+        if not all(os.path.isfile(p) for p in path):
+            raise exception.InvalidConfigException(
+                "The path of bbcp are not valid: {path}", path=path)
```

Comment:
Maybe "Invalid path '{path}' specified in `[bbcp] path` config option: {error_message}"?